snap/info_test.go

```diff
@@ -1206,11 +1204,26 @@ func (s *infoSuite) testInstanceDirAndFileMethods(c *C, info snap.PlaceInfo) {
 	c.Check(info.CommonDataSaveDir(), Equals, "/var/lib/snapd/save/snap/name_instance")
 	c.Check(info.UserXdgRuntimeDir(12345), Equals, "/run/user/12345/snap.name_instance")
 	// XXX: Those are actually a globs, not directories
-	c.Check(info.DataHomeDir(nil), Equals, "/home/*/snap/name_instance/1")
-	c.Check(info.CommonDataHomeDir(nil), Equals, "/home/*/snap/name_instance/common")
 	c.Check(info.XdgRuntimeDirs(), Equals, "/run/user/*/snap.name_instance")
 }
 
+func (s *infoSuite) TestDirMultipleHomeDirs(c *C) {
+	dirs.SetRootDir("")
+	dirs.SetSnapHomeDirs("/home,/home/group1,/home/group2,/home/group3")
+	info := &snap.Info{SuggestedName: "name"}
+	info.SideInfo = snap.SideInfo{Revision: snap.R(1)}
+	s.testDataHomeDirsSlice(c, info)
+}
+
+func (s *infoSuite) testDataHomeDirsSlice(c *C, info snap.PlaceInfo) {
+	homedirs := []string{"/home/*/snap/name/1", "/home/group1/*/snap/name/1", "/home/group2/*/snap/name/1", "/home/group3/*/snap/name/1"}
+	commonHomedirs := []string{"/home/*/snap/name/common", "/home/group1/*/snap/name/common", "/home/group2/*/snap/name/common", "/home/group3/*/snap/name/common"}
+	for i := range homedirs {
+		c.Check(info.DataHomeDirs(nil)[i], Equals, homedirs[i])
```

Comment:
it would be a good idea to try the test once with `opts.HiddenSnapDataDir` set and unset.